request-manager/test/db/db.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2017, Square, Inc.
+
+package db
+
+import (
+	"database/sql"
+	"fmt"
+	"os/exec"
+	"strings"
+	"sync"
+
+	"github.com/go-sql-driver/mysql"
+	"github.com/square/spincycle/config"
+	rmtest "github.com/square/spincycle/request-manager/test"
+	"github.com/square/spincycle/test"
+)
+
+type manager struct {
+	defaultDSNConfig *mysql.Config
+	dbType           string
+	defaultCmdArgs   []string
+	cliPath          string
+	//
+	dbs map[string]*sql.DB // db name => *sql.Db
+	*sync.Mutex
+}
+
+// A Manager manages the lifecycle of test databases. It is safe to use
+// with multiple databases concurrently.
+type Manager interface {
```

Comment:
Good start on this. This is something we'll need to make db testing simple. I've been thinking that we might just want to factor out and abstract the whole concept, because we could use it for Etre and other various projects, too. But first, I'll look around and see if there's already a reusable solution. If not, I think we can make a good contribution to the Go+MySQL world with such a pkg, time permitting.